app/src/main/java/org/mozilla/focus/utils/FirebaseHelper.java

```diff
@@ -78,223 +76,115 @@
     static final String STR_SHARE_APP_DIALOG_CONTENT = "str_share_app_dialog_content";
     static final String STR_SHARE_APP_DIALOG_MSG = "str_share_app_dialog_msg";
 
-    private HashMap<String, Object> remoteConfigDefault;
-    private static boolean changing = false;
-    private static Boolean pending = null;
+    static final String NEWLINE_PLACE_HOLDER = "<BR>";
 
-    @Nullable
-    private static BlockingEnablerCallback enablerCallback;
+    @SuppressFBWarnings(value = "MS_CANNOT_BE_FINAL", justification = "Abstract class can be replaced with an empty implementation. But I'm not determined to do it.")
+    public static FirebaseContract firebaseContract;
 
     // the file name to used when you want to set the default value of RemoteConfig
     private static final String REMOTE_CONFIG_JSON = "remote_config.json";
 
     private FirebaseHelper() {
     }
 
-    // inject delay to BlockingEnabler
-    @VisibleForTesting
-    public static void injectEnablerCallback(BlockingEnablerCallback callback) {
-        enablerCallback = callback;
-    }
-
-    public static void init(final Context context, boolean enabled) {
+    public static void init(final Context context, boolean enabled, FirebaseContract contract) {
 
-        if (getInstance() == null) {
-            initInternal(new FirebaseHelper());
+        if (firebaseContract == null) {
+            firebaseContract = contract;
         }
-        bind(context, enabled);
+        firebaseContract.setRemoteConfigDefault(contract.getRemoteConfigDefault());
```

Comment:
From the definition of the FirebaseContract, any of its instances will have their own default config during construction. 
Based on the above statement, in the case when firebaseContract is null, the call to setRemoteConfigDefault seems redundant.
On the other hand, when firebaseContract is not null, it's possible the new contract and the previous one is of different implementations (the logic here allows this possibility), overriding the original config with config from another contract seems weird to me.

Reply:
fixed